devel/export-map.lua

```diff
@@ -0,0 +1,332 @@
+        end
+
+        KEYS.MATERIAL.SOLID = {} -- everything but plants (stones, gems, metals)
+        KEYS.MATERIAL.METAL = {}
+        KEYS.MATERIAL.STONE = {}
+        KEYS.MATERIAL.GEM = {}
+
+        for id, rock in ipairs(df.global.world.raws.inorganics) do
+            local material = rock.material
+            local name = material.state_adj.Solid
+            KEYS.MATERIAL.SOLID[id] = name
+-- cant sort by key see
+-- https://stackoverflow.com/questions/26160327/sorting-a-lua-table-by-key
+            KEYS.MATERIAL.STONE[id] = material.flags.IS_STONE and name or false
+            KEYS.MATERIAL.GEM[id] = material.flags.IS_GEM and name or false
+            KEYS.MATERIAL.METAL[id] = material.flags.IS_METAL and name or false
+        end
+    end
+
+    if(options.liquid) then
+        KEYS.LIQUID = liquid_list
+    end
+
+    if(options.flow) then
+        KEYS.FLOW = {}
+        for i=0, 7 do
+            KEYS.FLOW[i] = i
+        end
+    end
+
+    return KEYS
+end
+
+local function export_all_z_levels(fortress_name, folder, options)
+    local xmax, ymax, zmax = dfhack.maps.getTileSize()
+    local filename = string.format("%s/%s.json", folder, fortress_name)
+
+    if dfhack.filesystem.exists(filename) then
+        qerror('Destination file ' .. filename .. ' already exists!')
+        return false
+    end
+
+    local data = {}
+
+    data.ARGUMENT_OPTION_ORDER = options
+    data.MAP_SIZE = {
+        x = xmax,
+        y = ymax,
+        -- subtract underworld levels if excluded from options
+        z = include_underworld_z and zmax or (zmax - underworld_z),
+        underworld_z_level = include_underworld_z and underworld_z or nil,
+        evilness = evilness or nil,
+    }
+    data.KEYS = setup_keys(options)
+
+    data.map = {}
+
+    local zmin = 0
+    if not include_underworld_z then -- skips all z-levels in the underworld
+        zmin = underworld_z
+    end
+
+    -- start from bottom z-level (underworld) to top z-level (sky)
+    for z = zmin, zmax do
+        local level_data = {}
+        for y = 0, ymax - 1 do
+            local row_data = {}
+            for x = 0, xmax - 1 do
+                local classification = classify_tile(options, x, y, z)
+                table.insert(row_data, classification)
+            end
+            table.insert(level_data, row_data)
+        end
+        table.insert(data.map, level_data)
+    end
+
+    local f = assert(io.open(filename, 'w'))
+    f:write(json.encode(data))
+    f:close()
+    print("File created in Dwarf Fortress folder under " .. filename)
+end
+
+
+local function export_fortress_map(options)
+    local fortress_name = dfhack.TranslateName(
+        df.global.world.world_data.active_site[0].name
+    )
+    local export_path = "map-exports/" .. fortress_name
+    dfhack.filesystem.mkdir_recursive(export_path)
+    export_all_z_levels(fortress_name, export_path, options)
+end
+
+if dfhack_flags.module then
+    return
+end
```

Comment:
this script is not declared as a module, so this does nothing

Reply:
Removed via https://github.com/DFHack/scripts/pull/1329/commits/eac50825c0f59f457e725d8c17a528f882a67759